components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/InMemoryAPIDeployer.java

```diff
@@ -58,6 +62,37 @@ public boolean deployAPI(String apiId, String gatewayLabel) {
         return false;
     }
 
+    public boolean deployAllAPIsAtGatewayStartup (Set<String> assignedGatewayLabels) {
+
+        if (gatewayArtifactSynchronizerProperties.isRetrieveFromStorageEnabled()) {
+            if (artifactRetriever != null) {
+                try {
+                    for (Iterator<String> it = assignedGatewayLabels.iterator(); it.hasNext();) {
+                        String label = it.next();
+                        List<String> gatewayRuntimeArtifacts = ServiceReferenceHolder
+                                .getInstance().getArtifactRetriever().retrieveAllArtifacts(label);
+                        for (String APIruntimeArtifact :gatewayRuntimeArtifacts){
+                            GatewayAPIDTO gatewayAPIDTO = null;
+                            try {
+                                gatewayAPIDTO = new Gson().fromJson(APIruntimeArtifact, GatewayAPIDTO.class);
+                                apiGatewayAdmin.deployAPI(gatewayAPIDTO);
+                            } catch (AxisFault axisFault) {
+                                log.error("Error in deploying" + gatewayAPIDTO.getName()+ " to the Gateway ");
```

Comment:
The error stack trace is swallowed here. Don't we need to log it, since we do not throw the error?

Reply:
https://github.com/wso2/carbon-apimgt/pull/9546/commits/ad587f2f16bdef1c6b05054c42b780dda8e218b9